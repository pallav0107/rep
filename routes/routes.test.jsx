import React from 'react';
import { shallow } from 'enzyme';
import Routes from './routes';
import { Route } from 'react-router';
import {
  Error,
  CustomerProfile,
  CustomerSearch,
  DashBoard,
  CreateCustomer,
  ProfileDetails,
} from '../layouts';
import "../setupTests"

describe('Routes', () => {
  it('should render routes', () => {
    const component = <Routes />;
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct routes', () => {
    const wrapper = shallow(<Routes />);
    const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});
    expect(pathMap['/home']).toBe(DashBoard);
    expect(pathMap['/create']).toBe(CreateCustomer);
    expect(pathMap['/search']).toBe(CustomerSearch);
    expect(pathMap['/profiledata']).toBe(ProfileDetails);
    expect(pathMap['/']).toBe(DashBoard);
  });
});
