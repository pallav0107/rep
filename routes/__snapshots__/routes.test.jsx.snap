// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render routes 1`] = `
<BrowserRouter
  basename="/"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/userhome"
    />
    <Redirect
      from="/callback"
      push={false}
      to="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
